src/lib/transport.js

```diff
@@ -0,0 +1,190 @@
+'use strict'
+
+const Packet = require('./packet')
+const moment = require('moment')
+const cryptoHelper = require('../utils/crypto')
+const cc = require('../utils/condition')
+const co = require('co')
+const debug = require('debug')('ilp:transport')
+const assert = require('assert')
+const base64url = require('../utils/base64url')
+const BigNumber = require('bignumber.js')
+
+function createPacketAndCondition ({
+  destinationAmount,
+  destinationAccount,
+  secret,
+  data,
+  expiresAt,
+  protocol
+}) {
+  assert(typeof destinationAmount === 'string', 'destinationAmount must be a string')
+  assert(typeof destinationAccount === 'string', 'destinationAccount must be a string')
+  assert(Buffer.isBuffer(secret), 'secret must be a buffer')
+
+  const id = base64url(cryptoHelper.getReceiverId(secret))
+  const address = destinationAccount + '.~' + protocol + '.' + id
+  const blob = data && base64url(cryptoHelper.aesEncryptObject(data, secret))
+  const packet = Packet.serialize({
+    amount: destinationAmount,
+    address: address,
+    expiresAt: expiresAt,
+    data: { blob }
+  })
+
+  const condition = base64url(cc.toCondition(
+    cryptoHelper.hmacJsonForPskCondition(
+      packet,
+      secret)))
+
+  return {
+    packet,
+    condition
+  }
+}
+
+function _reject (plugin, id, reason) {
+  return plugin
+    .rejectIncomingTransfer(id, reason)
+    .then(() => Promise.reject(new Error(reason)))
+}
+
+function listen (plugin, {
+  // TODO: best way to do receiverId?
+  secret,
+  allowOverPayment
+}, callback, protocol) {
+  assert(plugin && typeof plugin === 'object', 'plugin must be an object')
+  assert(typeof callback === 'function', 'callback must be a function')
+  assert(Buffer.isBuffer(secret), 'opts.secret must be a buffer')
+
+  // TODO: do this async
+  // yield plugin.connect()
```

Comment:
I'll change listen to be async as per your other comment, and then I'll be able to connect the plugin